site/app/views/admin/DockerView.php

```diff
@@ -18,289 +18,24 @@ public function displayDockerPage(array $docker_data) {
-                if ($is_match) {
-                    $machine_system_details[$current_machine]["disk"] = $matches[1];
-                }
-
-                $is_match = preg_match("/System Load: \((.+)\)/", $buffer, $matches);
-                if ($is_match) {
-                    $machine_system_details[$current_machine]["load"] = $matches[1];
-                }
-
-                $is_match = preg_match("/^\[Last ran on: (.+)\]/", $buffer, $matches);
-                if ($is_match) {
-                    $sysinfo_last_updated = $matches[1];
-                }
-
-                $is_match = preg_match("/ERR:/", $buffer, $matches);
-                if ($is_match) {
-                    $error_logs[] = "Failed to update system info";
-                }
-
-                if (preg_last_error() != PREG_NO_ERROR) {
-                    $error_logs[] = "Error while parsing system info logs";
-                    break;
-                }
-
-                $buffer = strtok("\n");
-            }
-        }
-
-        $no_image_capabilities = array_unique($no_image_capabilities);
-        foreach ($aliases as &$alias) {
-            $alias = array_unique($alias);
-        }
         return $this->output->renderTwigTemplate(
             "admin/Docker.twig",
             [
-                "autograding_containers" => $docker_data['autograding_containers'],
-                "capabilities" => $capabilities,
-                "worker_machines" => $worker_machines,
-                "no_image_capabilities" => $no_image_capabilities,
-                "image_to_capability" => $image_to_capability,
-                "capability_to_color" => $capability_to_color,
+                "autograding_containers" => $docker_ui->getAutogradingContainers(),
+                "capabilities" => $docker_ui->getCapabilities(),
+                "worker_machines" => $docker_ui->getWorkerMachines(),
+                "no_image_capabilities" => $docker_ui->getNoImageCapabilities(),
+                "capability_to_color_mapping" => $docker_ui->getCapabilityToColorMapping(),
                 "admin_url" => $this->core->buildUrl(["admin"]),
-                "last_updated" => $last_ran,
-                "sysinfo_last_updated" => $sysinfo_last_updated,
-                "machine_to_update" => $machine_to_update,
-                "image_info" => $image_info,
-                "machine_docker_version" => $machine_docker_version,
-                "machine_system_details" => $machine_system_details,
-                "aliases" => $aliases,
+                "last_updated" => $docker_ui->getLastRan(),
+                "sysinfo_last_updated" => $docker_ui->getSysinfoLastUpdated(),
+                "docker_images" => $docker_ui->getDockerImages(),
+                // "machine_to_update" => $machine_to_update,
+                // "image_info" => $image_info,
+                //"machine_docker_version" => $machine_docker_version,
+                //"machine_system_details" => $machine_system_details,
```

Comment:
These will presumably be removed before the PR is merged, right?

Reply:
Oops yup, old data no longer being used